# Pandas Library 
Pandas is the most popular library of python used for data analysis.
To work with it, first import the library.(you can use alias for simplicity)

In [1]:
import pandas as pd

Two main concepts in pandas are:

1. DataFrame
2. Series

We will create our own data using these as follows

**DataFrame** <br>
DataFrame is a table. It contains an array of entries. Each entry corresponds to row and coloumn.
For example: 31 is a entry whose row=0, column = yes

In [2]:
data = pd.DataFrame({'yes':[31,3],'no':[33,9]})
print(data)

   yes  no
0   31  33
1    3   9


As you can see, the table is automatically given index values which starts for 0.
If you wish to use give specific values to index, do  follows:

In [3]:
data = pd.DataFrame({'Bob':["I likes it","Bland"],
                     'Alice':["It was awful","I liked it"]},
                     index = ['Product A','Product B'])
print(data)

                  Bob         Alice
Product A  I likes it  It was awful
Product B       Bland    I liked it


**Series**:
Series is a sequence of data values. If DataFrame is Table, series is a List.

In [4]:
data = pd.Series([1,2,3,4])
data

0    1
1    2
2    3
3    4
dtype: int64

Series is just a colum of DataFrame. So, you can assign a row lable using `index` parameter. However, a Series does not have a column name. It only has one name overall.

In [5]:
data = pd.Series([30,40,50],index=['Subject A','Subject B','Subject c'],name = 'Marks of a Student')
data

Subject A    30
Subject B    40
Subject c    50
Name: Marks of a Student, dtype: int64

### Reading data files 

As of now, we have seen how to create our own data. But, most of the time, we won't be creating data by hand. Instead, we will be working with the already existing data.
Data can be stored in any format.By far the most used is CSV files. <br>
CSV stands for Comma separeted values.
When you open a CSV file, you will get to see the data values separated by commas and hence the name CSV. <br>
To read the data into DataFrame, we use `pd.read_csv()` function 

In [6]:
data = pd.read_csv("/kaggle/input/datasets/spoortibadiger/network-intrusion-dataset/Wednesday-workingHours.pcap_ISCX.csv")

Before we move further, let us know how to upload the datasets to kaggle and sue them:
1. **Add the dataset (if not already done)** <br>
In your notebook, look at the right-hand sidebar (in the Notebook editor). Click "Add Input" (sometimes shown as a "+" next to "Input"). Search for your dataset by name and click the + to add it. This mounts it — it doesn't copy files into your working code.
2. **Find the exact path** <br>
Once added, Kaggle places datasets under:<br>
`/kaggle/input/<dataset-folder-name>/`
3. To see the exact folder and file names (they don't always match what you expect), run this in a code cell: <br>
```
import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))
```
Now use the path you get after running this code in the `read_csv()`

We can use the `shape` attribute to check how large the resulting DataFrame is:

In [7]:
data.shape

(692703, 79)

The above output tells us that the new DataFrame created have total of 692703 records split across 79 columns.
<br>
To examine the content of the dataset, we can use `head()` command which gives us the first 5 rows of the dataset.